static/js/lib/coupon.js

```diff
@@ -89,3 +90,21 @@ export const _calculateRunPrice = (
 // allow mocking of function
 export { _calculateRunPrice as calculateRunPrice };
 import { calculateRunPrice } from './coupon';
+
+export function makeAmountMessage(coupon: Coupon): string {
+  if (coupon.amount_type === COUPON_AMOUNT_TYPE_FIXED_DISCOUNT) {
+    return `$${coupon.amount} off`;
+  } else if (coupon.amount_type === COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT) {
+    return `${coupon.amount.times(100).toDecimalPlaces(0).toString()}% off`;
+  } else {
+    return '';
+  }
```

Comment:
this might be a little more readable with a `switch` statement (debateable I guess, up to you!)